tools/run_tests/artifacts/build_artifact_python.bat

```diff
@@ -32,7 +32,10 @@ set GRPC_PYTHON_BUILD_WITH_CYTHON=1
 
 @rem Allow build_ext to build C/C++ files in parallel
 @rem by enabling a monkeypatch. It speeds up the build a lot.
-set GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS=4
+@rem Use externally provided GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS value if set.
+if "%GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS%"=="" (
+  set GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS=4
```

Comment:
Yes, it's somewhat relevant. I added the task_runner.py's `--inner_jobs` arg to the invocation in grpc_build_artifacts.bat, so I'm also adjusting python to respect that setting (instead of just hardcoding value 4).

Ad default value 2 vs 4 - good point but it doesn't really matter much, since --inner_jobs is now set. I guess you can interpret it as python artifact being slower, so I set the default higher, for C# the values has smaller effect on overall duration (or at least it had before I switched the build to ninja).